driver-core/src/main/com/mongodb/MongoQueryException.java

```diff
@@ -51,43 +62,33 @@ public MongoQueryException(final ServerAddress address, final int errorCode, fin
      * @param errorCodeName the error code name
      * @param errorMessage the error message
      * @since 4.6
+     * @deprecated Prefer {@link #MongoQueryException(BsonDocument, ServerAddress)}
      */
-    public MongoQueryException(final ServerAddress address, final int errorCode, final @Nullable String errorCodeName,
+    @Deprecated
+    public MongoQueryException(final ServerAddress address, final int errorCode, @Nullable final String errorCodeName,
             final String errorMessage) {
-        super(errorCode, errorCodeName,
-                format("Query failed with error code %d with name '%s' and error message '%s' on server %s", errorCode, errorCodeName,
-                        errorMessage, address),
-                address);
-        this.errorMessage = errorMessage;
+        this(manufactureResponse(errorCode, errorCodeName, errorMessage), address);
     }
 
     /**
      * Construct an instance from a command exception.
      *
      * @param commandException the command exception
      * @since 3.7
+     * @deprecated Prefer {@link #MongoQueryException(BsonDocument, ServerAddress)}
      */
+    @Deprecated
     public MongoQueryException(final MongoCommandException commandException) {
-        this(commandException.getServerAddress(), commandException.getErrorCode(), commandException.getErrorCodeName(),
-                commandException.getErrorMessage());
-        addLabels(commandException.getErrorLabels());
+        this(commandException.getResponse(), commandException.getServerAddress());
     }
 
-    /**
-     * Gets the error code for this query failure.
-     *
-     * @return the error code
-     */
-    public int getErrorCode() {
-        return getCode();
-    }
-
-    /**
-     * Gets the error message for this query failure.
-     *
-     * @return the error message
-     */
-    public String getErrorMessage() {
-        return errorMessage;
+    private static BsonDocument manufactureResponse(final int errorCode, @Nullable final String errorCodeName, final String errorMessage) {
```

Comment:
Necessary to manufacture the response document to support the deprecated constructors.  In practice this will go unused since the driver doesn't use the deprecated constructors, but application tests may do so.